scripts/zulip_emoji_merge_delegate.py

```diff
@@ -50,7 +50,25 @@
 print(f"public_response:{public_response}")
 print(f"reviewers_response:{reviewers_response}")
 
-messages = (public_response['messages']) + (reviewers_response['messages'])
+# Get the oldest message in any topic in `PR reviews` whose title matches `#{PR_NUMBER}`.
+first_message_in_PR_reviews = client.get_messages({
+    "anchor": "oldest",
+    "num_before": 0,
+    "num_after": 1,
+    "narrow": [
+        {"operator": "channel", "operand": "PR reviews"},
+        {"operator": "search", "operand": f'#{PR_NUMBER}'},
+    ],
+})
+print(f'First message:{first_message_in_PR_reviews}')
```

Comment:
Where do you actually check that the title contains the PR number? It looks like you're only checking the whole message at once.

Reply:
You were right again.  I changed the strategy a bit: I loop through the initial filter of all (public) messages that contain `PR number` somewhere and I extract the ID of the first time that a message in `PR reviews` has a topic matching the `PR number`.

Using this ID, the logic then stays the same: react to all messages that contain `PR number` in their body and also the message whose ID has been singled out (if it exists, I assume, I do not know python enough to know what happens if the `specialID` variable is not set).